lib/kramdown/converter/syntax_highlighter/rouge.rb

```diff
@@ -27,7 +27,16 @@ def self.call(converter, text, lang, type, call_opts)
       lexer = ::Rouge::Lexer.find_fancy(lang || opts[:default_lang], text)
       return nil if opts[:disable] || !lexer
 
-      formatter = (opts.fetch(:formatter, ::Rouge::Formatters::HTML)).new(opts)
+      begin
+        # Rouge::Formatters::HTMLLegacy is available on Rouge 2.0 or later
+        formatter_class = ::Rouge::Formatters::HTMLLegacy
+        opts[:css_class] ||= 'highlight'
+      rescue NameError
+        # Fallback to Rouge 1
+        formatter_class = ::Rouge::Formatters::HTML
+      end
```

Comment:
I am not sure if I properly understand what you mean.

Do you mean that `begin ... rescue ... end` should be placed like [these ones](https://github.com/5t111111/kramdown/blob/d0e81c3ee1531e5723af8e9805cd33181f984665/lib/kramdown/converter/syntax_highlighter/rouge.rb#L15-L22) like the below snippet? 

```ruby

...

  module Rouge

...

    begin
      FORMATTER_CLASS = ::Rouge::Formatters::HTMLLegacy
    rescue NameError
      FORMATTER_CLASS = ::Rouge::Formatters::HTML
    end

    def self.call(converter, text, lang, type, call_opts)

      ...

      formatter = (opts.fetch(:formatter, FORMATTER_CLASS)).new(opts)
      formatter.format(lexer.lex(text))
    end

...

  end

```